pr_insight/algo/ai_handlers/litellm_ai_handler.py

```
@@ -185,14 +191,30 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
-                "messages": messages,
-                "temperature": temperature,
-                "timeout": get_settings().config.ai_timeout,
-                "api_base": self.api_base,
-            }
+            # Currently O1 does not support separate system and user prompts
+            O1_MODEL_PREFIX = 'o1-'
+            model_type = model.split('/')[-1] if '/' in model else model
+            if model_type.startswith(O1_MODEL_PREFIX):
+                user = f"{system}\n\n\n{user}"
+                system = ""
+                get_logger().info(f"Using O1 model, combining system and user prompts")
```

Ruff

f-string without any placeholders

(F541)